File: jv-service/src/main/java/com/lcaohoanq/jvservice/domain/user/Address.java
package com.lcaohoanq.jvservice.domain.user;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "user_addresses")
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class Address {

    @Id
    @SequenceGenerator(name = "user_addresses_seq", sequenceName = "user_addresses_id_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "user_addresses_seq")
    @Column(name = "id", unique = true, nullable = false)
    @JsonProperty("id")
    private Long id;

    private String nameOfUser;

    private String phoneNumber;

    private String address;

    private boolean isDefault;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;


}

File: auth-service/src/main/kotlin/com/lcaohoanq/authservice/domains/user/Address.kt
package com.lcaohoanq.authservice.domains.user

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.*

@Entity
@Table(name = "user_addresses")
class Address(
    @Id
    @SequenceGenerator(
        name = "user_addresses_seq",
        sequenceName = "user_addresses_id_seq",
        allocationSize = 1
    )
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "user_addresses_seq")
    @Column(name = "id", unique = true, nullable = false)
    @JsonProperty("id")
    val id: Long? = null,
    val nameOfUser: String = "",
    val phoneNumber: String = "",
    val address: String = "",
    val isDefault: Boolean = false,
    val userId: Long,

) {
}